src/providerselector.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from "react";
+import {GitHub} from './gitProviders/github'
+import {UserConfig} from "./interfaces/UserConfig";
+import * as Helpers from './helpers'
+import {GitLab} from "./gitProviders/gitlab";
+
+export class ProviderSelector extends React.Component<any, any>{
+
+    constructor(props) {
+        super(props);
+        console.log(this.props);
+        this.state = {
+            'baseUrl': "",
+            'token': "",
+            'projectId': null,
+            'configFileName': "",
+            'branch': "",
+            'branchOptions': []
+        };
+
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    async componentDidMount() {
+        const config: UserConfig = await Helpers.loadConfig(this.props.context);
+        this.setState(config);
+    }
+
+    private handleChange(event) {
+        const { target: { name, value } } = event;
+        console.debug("Update state property: ", name, value);
+        this.setState({[name]: value});
+    }
+
+    const // @ts-ignore
```

Comment:
Same thing as in Github provider: do we need this empty declaration?

Reply:
Same instance as other one